src/composeappmanager.cc

```diff
@@ -299,22 +303,32 @@ data::InstallationResult ComposeAppManager::install(const Uptane::Target& target
   if (!cur_apps_to_fetch_and_update_.empty()) {
     res.description += "\n# Apps installed:";
   }
-  for (const auto& pair : cur_apps_to_fetch_and_update_) {
-    LOG_INFO << "Installing " << pair.first << " -> " << pair.second;
-    const bool just_install = res.result_code == data::ResultCode::Numeric::kNeedCompletion;
-    // I have no idea via the package manager interface method install() is const which is not a const
-    // method by its definition/nature
-    auto& non_const_app_engine = (const_cast<ComposeAppManager*>(this))->app_engine_;
-    auto run_res = just_install ? non_const_app_engine->install({pair.first, pair.second})
-                                : non_const_app_engine->run({pair.first, pair.second});
-    if (!run_res) {
-      res = data::InstallationResult(data::ResultCode::Numeric::kInstallFailed, "Could not install app");
-    } else {
-      res.description += "\n" + pair.second;
+  bool prune_images{cfg_.docker_prune};
+  const bool just_install = res.result_code == data::ResultCode::Numeric::kNeedCompletion;  // system update
+
+  if (!just_install || cfg_.create_containers_before_reboot) {
+    for (const auto& pair : cur_apps_to_fetch_and_update_) {
+      LOG_INFO << "Installing " << pair.first << " -> " << pair.second;
+      // I have no idea via the package manager interface method install() is const which is not a const
+      // method by its definition/nature
+      auto& non_const_app_engine = (const_cast<ComposeAppManager*>(this))->app_engine_;
+      bool run_res = just_install ? non_const_app_engine->install({pair.first, pair.second})
+                                  : non_const_app_engine->run({pair.first, pair.second});
+
+      if (!run_res) {
+        res = data::InstallationResult(data::ResultCode::Numeric::kInstallFailed, "Could not install app");
+      } else {
+        res.description += "\n" + pair.second;
+      }
     }
-  };
+  } else {
+    LOG_INFO << "Apps' containers will be re-created and started just after succesfull boot on the new ostree version";
```

Comment:
should we add this into res.description to provide more install details to our backend?

Reply:
makes sense, will do it.

Reply:
updated.